eventhubs/client/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/ProxyConfiguration.java

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.net.PasswordAuthentication;
+import java.net.Proxy;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * A configuration object for Proxy
+ */
+public class ProxyConfiguration implements AutoCloseable {
+    private final ClientLogger logger = new ClientLogger(ProxyConfiguration.class);
+
+    public static final String PROXY_USERNAME = "PROXY_USERNAME";
+    public static final String PROXY_PASSWORD = "PROXY_PASSWORD";
+
+    private final PasswordAuthentication credentials;
+    private final Proxy proxyAddress;
+    private final ProxyAuthenticationType authentication;
+
+    /**
+     * Gets the system defaults for proxy configuration and authentication.
+     */
+    public static final ProxyConfiguration SYSTEM_DEFAULTS = new ProxyConfiguration();
+
+    ProxyConfiguration() {
+        this.credentials = null;
+        this.proxyAddress = null;
+        this.authentication = null;
+    }
+    /**
+     * Creates a proxy configuration that uses the {@code proxyAddress} and authenticates with provided
+     * {@code username}, {@code password} and {@code authentication}.
+     *
+     * @param authentication Authentication method to preemptively use with proxy.
+     * @param proxyAddress Proxy to use. If {@code null} is passed in, then the system configured {@link java.net.Proxy}
+     * is used.
+     * @param username Optional. Username used to authenticate with proxy. If not specified, the system-wide
+     * {@link java.net.Authenticator} is used to fetch credentials.
+     * @param password Optional. Password used to authenticate with proxy.
+     *
+     * @throws NullPointerException if {@code authentication} is {@code null}.
+     * @throws IllegalArgumentException if {@code authentication} is {@link ProxyAuthenticationType#BASIC} or
+     * {@link ProxyAuthenticationType#DIGEST} and {@code username} or {@code password} are {@code null}.
+     */
+    ProxyConfiguration(ProxyAuthenticationType authentication, Proxy proxyAddress, String username, String password) {
+        Objects.requireNonNull(authentication);
+        this.authentication = authentication;
+        this.proxyAddress = proxyAddress;
+
+        if (username != null && password != null) {
+            this.credentials = new PasswordAuthentication(username, password.toCharArray());
+        } else {
+            logger.asInfo().log("Username or password is null. Using system-wide authentication.");
+            this.credentials = null;
+        }
+    }
+
+    /**
+     * Gets the proxy authentication type
```

Comment:
nit: full stops are missing on some of these one-sentence JavaDocs

Reply:
Fixed.